### PR TITLE
Improved controller tests for assignments

### PR DESCRIPTION
**Pull request by katafira**
@sergio-ocon  @alexvkcr @delaluzparra

### CODE REVIEW

File: spec/controllers/assignments_controller_spec.rb [hunk 1]
```diff
@@ -1,15 +1,63 @@
...
+      it "should not allow a shift without end_time"  do
+        shift.end_time = nil
+        expect(shift).not_to be_valid
+      end
+      it "should not allow a shift without prebooked"  do
+        expect{
+
+          shift.prebooked = nil 
+          shift.valid?
+          }.to raise_exception("nil can't be coerced into Fixnum")
```
Comment (sergio-ocon) [inline spec/controllers/assignments_controller_spec.rb]: This does not work. 
expected Exception with "nil can't be coerced into Fixnum" but nothing was raised

File: spec/controllers/assignments_controller_spec.rb [hunk 2]
```diff
@@ -1,15 +1,63 @@
 require 'rails_helper'
 
 RSpec.describe AssignmentsController, type: :controller do
-  context "invalid offspring and/or shift" do
-    pending
+  context "invalid " do
+    let(:offspring) { FactoryGirl.build(:offspring)}
+    let(:shift) { FactoryGirl.build(:shift)}
+    
+    describe "Offspring," do
```
Comment (sergio-ocon) [inline spec/controllers/assignments_controller_spec.rb]: This is not a controller spec.
This is a model spec.

Comment (sergio-ocon) [inline spec/controllers/assignments_controller_spec.rb]: Modify this into a model spec, and there you can add things.
This looks like a typo on my side that you followed

**Comment by sergio-ocon** [comment]
Please run rubocop in your code to test for sintaxis

File: spec/controllers/assignments_controller_spec.rb (hunk 1 above)
Comment (alexvkcr) [inline spec/controllers/assignments_controller_spec.rb]: This spec will work when the Shift model from #4 is included in this branch. 
Sorry about not having included it in the comments.

**Comment by sergio-ocon** [comment]
You need to rebase before we can merge this.

File: spec/controllers/assignments_controller_spec.rb [hunk 3]
```diff
@@ -1,28 +1,70 @@
...
+        shift.start_time = nil
+        expect(shift).not_to be_valid
+      end
+      it "should not allow a shift without end_time" do
+        shift.end_time = nil
+        expect(shift).not_to be_valid
+      end
+      it "should not allow a shift without prebooked" do
+        expect{
+          shift.prebooked = nil
```
Comment (alexvkcr) [inline spec/controllers/assignments_controller_spec.rb]: Rubocop says this multi-line block should be avoided, any suggestion on how to perform this action?

**Comment by alexvkcr** [comment]
Regargind a comment I posted a time ago.

'This spec will work when the Shift model from #4 is included in this branch. 
Sorry about not having included it in the comments.'

I just came up with an idea. Should we do a PR with just the update of the model, merge to your repository and then rebase again? Or instead we keep waiting until #8 is merged?

File: spec/controllers/assignments_controller_spec.rb (hunk 3 above)
Comment (sergio-ocon) [inline spec/controllers/assignments_controller_spec.rb]: expect(shift.prebooked).to be_nil,

but I don't understand why are you expecting if you didn't change anything before

File: spec/controllers/assignments_controller_spec.rb [hunk 4]
```diff
@@ -1,28 +1,70 @@
...
+      it "should not allow a shift without start_time" do
+        shift.start_time = nil
+        expect(shift).not_to be_valid
+      end
+      it "should not allow a shift without end_time" do
+        shift.end_time = nil
+        expect(shift).not_to be_valid
+      end
+      it "should not allow a shift without prebooked" do
+        expect{
```
Comment (sergio-ocon) [inline spec/controllers/assignments_controller_spec.rb]: You should assign before doing the expect

**Comment by alexvkcr** [comment]
There is one failing test, 'AssignmentsController invalid  Shift, should not allow a shift without prebooked'

It should work when PR #8 is merged because #8 includes changes in shift model.

**Comment by sergio-ocon** [comment]
?

In your code you are adding shift.errors.add (:shift,....)
That is wrong, you don't need to specify :shift. But I don't see those changes in the github interface...

**Comment by sergio-ocon** [comment]
This is an example of an error that you make on refactoring because you are not testing it.

**Comment by alexvkcr** [comment]
Ready to review, I can't assign it to anybody else as we do not have permission anymore

**Comment by alexvkcr** [comment]
Please anyone review @katafira @agarciavera @sebjimenez

**Comment by katafira** [comment]
Revised, it's not ready to merge yet. I'll keep an eye on it as you advance.

**Comment by alexvkcr** [comment]
EEEEyyyyy somebody, review or i will tell on :angry: jajaj

**Comment by sergio-ocon** [comment]
Change the name,
Take out the label

**Comment by katafira** [comment]
@sergio-ocon Changed the title. I think we already mentioned a while back that we are not allowed to change labels.

**Comment by sergio-ocon** [comment]
@katafira Then make a request here to change the label when you need it...

**Pull request merged**